presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -45,20 +58,50 @@ HttpResponse::~HttpResponse() {
 void HttpResponse::append(std::unique_ptr<folly::IOBuf>&& iobuf) {
   VELOX_CHECK(!iobuf->isChained());
   VELOX_CHECK(!hasError());
-  const uint64_t dataLength = iobuf->length();
-  void* buf{nullptr};
+
+  uint64_t dataLength = iobuf->length();
+  auto dataStart = iobuf->data();
+  if (!bodyChain_.empty()) {
+    auto tail = bodyChain_.back().get();
+    auto space = tail->tailroom();
```

Comment:
Thanks @xiaoxmeng . Some thoughts: Do you think in this scenario we can make use of AllocationPool? It looks similar in terms of what they do.